src/frontend/src/binder/expr/function.rs

```diff
@@ -258,31 +258,48 @@ impl Binder {
                 // We can safely unwrap here
                 let ast = parse_result.unwrap();
 
-                let mut clean_flag = true;
-
-                // We need to check if the `udf_context` is empty first, consider the following example:
-                // - create function add(INT, INT) returns int language sql as 'select $1 + $2';
-                // - create function add_wrapper(INT, INT) returns int language sql as 'select add($1, $2)';
-                // - select add_wrapper(1, 1);
-                // When binding `add($1, $2)` in `add_wrapper`, the input args are [$1, $2] instead of
-                // the original [1, 1], thus we need to check `udf_context` to see if the input
-                // args already exist in the context. If so, we do NOT need to create the context again.
-                // Otherwise the current `udf_context` will be corrupted.
+                let mut clean_flag = false;
```

Comment:
There is no need to maintain a `clean_flag`. The stash-and-restore method can achieve the same effect when the context is empty.

Reply:
Indeed, since the root binding udf's context will initially be empty, and will be restored eventually.